File: csharp/UrakawaToolkit/media/data/IManagedMedia.cs
using System;
using System.Collections.Generic;
using System.Text;
using urakawa.events.media;

namespace urakawa.media.data
{
	/// <summary>
	/// Common interface for <see cref="IMedia"/> that use <see cref="MediaData"/> to store their content
	/// </summary>
	public interface IManagedMedia : IMedia
	{
		/// <summary>
		/// Event fired after the <see cref="MediaData"/> of the <see cref="IManagedMedia"/> has changed
		/// </summary>
		event EventHandler<MediaDataChangedEventArgs> mediaDataChanged;
		
		/// <summary>
		/// Gets the <see cref="MediaData"/> storing the content
		/// </summary>
		/// <returns>The media data</returns>
		MediaData getMediaData();

		/// <summary>
		/// Sets the <see cref="MediaData"/> storing the content
		/// </summary>
		/// <param name="data">The new media data</param>
		/// <exception cref="exception.MethodParameterIsNullException">
		/// Thrown when <paramref name="data"/> is <c>null</c>
		/// </exception>
		void setMediaData(MediaData data);

		/// <summary>
		/// Gets the <see cref="MediaDataFactory"/> creating the <see cref="MediaData"/>
		/// used by the <see cref="IManagedMedia"/>.
		/// Convenience for <c>getMediaData().getMediaDataManager().getMediaDataFactory()</c>
		/// </summary>
		/// <returns>The media data factory</returns>
		MediaDataFactory getMediaDataFactory();
	}
}

File: java/core/src/org/daisy/urakawa/media/data/IManagedMedia.java
package org.daisy.urakawa.media.data;

import org.daisy.urakawa.exception.MethodParameterIsNullException;
import org.daisy.urakawa.media.IMedia;

/**
 * An media for which the data source is managed data {@link IMediaData}.
 * 
 * @leafInterface see {@link org.daisy.urakawa.LeafInterface}
 * @see org.daisy.urakawa.LeafInterface
 * @stereotype OptionalLeafInterface
 * @depend - Aggregation 1 org.daisy.urakawa.media.data.IMediaData
 * @depend - Clone - org.daisy.urakawa.media.data.IManagedMedia
 */
public interface IManagedMedia extends IMedia
{
    /**
     * @return the data object. Cannot be null.
     */
    public IMediaData getMediaData();

    /**
     * @param data
     *        cannot be null
     * @throws MethodParameterIsNullException
     *         NULL method parameters are forbidden
     * @tagvalue Events "MediaDataChanged"
     * @stereotype Initialize
     */
    public void setMediaData(IMediaData data)
            throws MethodParameterIsNullException;
}
